clusterdock/topologies/cdh/actions.py

```diff
@@ -215,3 +281,110 @@ def remove_files(cluster, files, nodes):
+    acl_data = acl_data.replace('EXAMPLE.COM', network.upper())
+    config_files[KDC_ACL_FILENAME] = acl_data
+
+    # Serialize them back to the kdc Docker container
+    for kdc_conf_file in KDC_CONFIG_FILES:
+        o_tarstream = io.BytesIO()
+        with tarfile.open(fileobj=o_tarstream, mode='w') as tarfile_:
+            encoded_file_contents = config_files[kdc_conf_file].encode()
+            tarinfo = tarfile.TarInfo(basename(kdc_conf_file))
+            tarinfo.size = len(encoded_file_contents)
+            tarfile_.addfile(tarinfo, io.BytesIO(encoded_file_contents))
+        o_tarstream.seek(0)
+        client.put_archive(container=kdc_node.container_id, path=dirname(kdc_conf_file), data=o_tarstream)
+
+def _setup_kdc_server(cluster, nodes, network, kerberos_principals):
+    logger.info('Setting up kdc server ...')
+    principal_cmds = ""
+    if kerberos_principals:
+        princ_list = kerberos_principals.split(',')
+        create_principals = " ".join(["kadmin.local -q \"addprinc -randkey %s@%s\"; " % (item, network)
+                                      for item in princ_list])
+        create_keytab = "kadmin.local -q \"xst -norandkey -k %s %s\"; " % (KDC_KEYTAB_FILENAME,
+                                                                           ' '.join(['%s@%s' % (item, network) for item in princ_list]))
+        principal_cmds = '{} {}'.format(create_principals, create_keytab)
+
+    kdc_commands = ("kdb5_util create -s -r {realm} -P kdcadmin; "
+                    "kadmin.local -q \"addprinc -pw {adminpw} admin/admin@{realm}\"; "
+                    "kadmin.local -q \"addprinc -pw {cldradminpw} {cldradmin}@{realm}\"; "
+                    "{create_principals_keytab} krb5kdc; "
+                    "kadmind; "
+                    "authconfig --enablekrb5 --update; "
+                    "service sshd restart; "
+                    "service krb5kdc restart; "
+                    "service kadmin restart"
+                   ).format(realm=network,
+                            adminpw='acladmin',
+                            create_principals_keytab=principal_cmds,
+                            cldradmin=CLDR_MGR_PRINCIPAL_USER,
+                            cldradminpw=CLDR_MGR_PRINCIPAL_PW
+                           )
+    cluster.ssh(kdc_commands, nodes)
+    # Gather keytab file and krb5.conf file in KERBEROS_VOLUME_DIR directory which is mounted on host.
+    if kerberos_principals:
+        gather_kerberos_files = 'chmod 644 {}; cp {} {};'.format(KDC_KEYTAB_FILENAME,
+                                                                 KDC_KRB5_CONF_FILENAME,
+                                                                 KERBEROS_VOLUME_DIR)
+        cluster.ssh(gather_kerberos_files, nodes)
+
+def _fix_for_hue_kerberos(cluster, nodes, network, hue_node_name):
+    logger.info('Applying fix for hue...')
+    kdc_hue_commands = ("kadmin.local -q \"modprinc -maxrenewlife 90day krbtgt/{realm}\"; "
```

Comment:
Same comment as above. In general, avoid creating long strings for commands since they're brittle. Lists with individual commands that can be joined later are easier to handle.